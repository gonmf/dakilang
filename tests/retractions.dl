
beta(A) :- test(A)~
listing
beta(A) :- test(A)~

listing
MyRule("john", "mary").
listing
MyRule("john", "mary")~

listing
MyRule("john", "mary").

listing

MyRule("john", "mary").
MyRule("john", "mary").

listing

retract 0

listing

retract 20

listing

retract -20

listing

d(X, Y, X) :- (e(X); e(Y); e(Z)), (f(X); f(Y); f(Z))~
listing

retract 3
listing

d(X, Y, X) :- (e(X); e(Y); e(Z)), (f(X); f(Y); f(Z))~
listing


d(X, Y, X) :- (e(X); e(Y); e(Z)), (f(X); f(Y); f(Z)).
d(X, Y, X) :- (e(X); e(Y); e(Z)), (f(X); f(Y); f(Z)).
listing

d(X, Y, X) :- (e(X); e(Y); e(Z)), (f(X); f(Y); f(Z))~
listing

d(X, Y, X) :- e(Y), f(X)~
listing

retract 4
listing
