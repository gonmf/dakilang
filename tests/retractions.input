
beta(A) :- test(A)~
beta(A) :- test(A)~

MyRule("john", "mary").
MyRule("john", "mary")~

MyRule("john", "mary").


listing

retract 1

listing

retract 20

retract -20

listing
