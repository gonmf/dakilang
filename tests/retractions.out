>
> beta(A) :- test(A)~
Clause not found

> beta(A) :- test(A)~
Clause not found

>
> MyRule("john", "mary").
> MyRule("john", "mary")~
Clause removed

>
> MyRule("john", "mary").
>
>
> listing
0: MyRule('john', 'mary').

>
> retract 1
Invalid clause index

>
> listing
0: MyRule('john', 'mary').

>
> retract 20
Invalid clause index

>
> retract -20
Invalid clause index

>
> listing
0: MyRule('john', 'mary').

